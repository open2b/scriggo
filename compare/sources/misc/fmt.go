//+build ignore

package main

import (
	"fmt"
)

func main() {
	a := 10
	b := 20

	fmt.Print(a + b)
	fmt.Print(*(&b))
	fmt.Print(a/b, a+b, []int{1, 2, 3}, a-b, "aaa")
	fmt.Print(interface{}("bbb"), 22+4, a+20, b-(a*10))
	fmt.Print()
	c := []string{"a", "b", "c"}
	d := new(int)
	fmt.Print(c, *d)
	fmt.Print(c)
	fmt.Print(*d)
	fmt.Print(*d, *d, 32, *d, c)
	for i := 0; i < 20; i++ {
		fmt.Print(i, i+20, i-20, i*i)
	}
}
