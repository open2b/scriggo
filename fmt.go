// Copyright (c) 2019 Open2b Software Snc. All rights reserved.
// https://www.open2b.com

// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package scrigo

import (
	"fmt"
	"reflect"
)

var _fmt = Package{
	"Errorf":     fmt.Errorf,
	"Formatter":  reflect.TypeOf((*fmt.Formatter)(nil)).Elem(),
	"Fprint":     fmt.Fprint,
	"Fprintf":    fmt.Fprintf,
	"Fprintln":   fmt.Fprintln,
	"Fscan":      fmt.Fscan,
	"Fscanf":     fmt.Fscanf,
	"Fscanln":    fmt.Fscanln,
	"GoStringer": reflect.TypeOf((*fmt.GoStringer)(nil)).Elem(),
	"Print":      fmt.Print,
	"Printf":     fmt.Printf,
	"Println":    fmt.Println,
	"Scan":       fmt.Scan,
	"ScanState":  reflect.TypeOf((*fmt.ScanState)(nil)).Elem(),
	"Scanf":      fmt.Scanf,
	"Scanln":     fmt.Scanln,
	"Scanner":    reflect.TypeOf((*fmt.Scanner)(nil)).Elem(),
	"Sprint":     fmt.Sprint,
	"Sprintf":    fmt.Sprintf,
	"Sprintln":   fmt.Sprintln,
	"Sscan":      fmt.Sscan,
	"Sscanf":     fmt.Sscanf,
	"Sscanln":    fmt.Sscanln,
	"State":      reflect.TypeOf((*fmt.State)(nil)).Elem(),
	"Stringer":   reflect.TypeOf((*fmt.Stringer)(nil)).Elem(),
}
