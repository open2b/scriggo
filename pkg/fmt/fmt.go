// Go version: go1.11.5

package fmt

import original "fmt"
import "scrigo"
import "reflect"

var Package = scrigo.Package{
	"Errorf": original.Errorf,
	"Formatter": reflect.TypeOf((*original.Formatter)(nil)).Elem(),
	"Fprint": original.Fprint,
	"Fprintf": original.Fprintf,
	"Fprintln": original.Fprintln,
	"Fscan": original.Fscan,
	"Fscanf": original.Fscanf,
	"Fscanln": original.Fscanln,
	"GoStringer": reflect.TypeOf((*original.GoStringer)(nil)).Elem(),
	"Print": original.Print,
	"Printf": original.Printf,
	"Println": original.Println,
	"Scan": original.Scan,
	"ScanState": reflect.TypeOf((*original.ScanState)(nil)).Elem(),
	"Scanf": original.Scanf,
	"Scanln": original.Scanln,
	"Scanner": reflect.TypeOf((*original.Scanner)(nil)).Elem(),
	"Sprint": original.Sprint,
	"Sprintf": original.Sprintf,
	"Sprintln": original.Sprintln,
	"Sscan": original.Sscan,
	"Sscanf": original.Sscanf,
	"Sscanln": original.Sscanln,
	"State": reflect.TypeOf((*original.State)(nil)).Elem(),
	"Stringer": reflect.TypeOf((*original.Stringer)(nil)).Elem(),
}
