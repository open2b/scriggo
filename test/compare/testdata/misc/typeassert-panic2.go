// paniccheck

package main

import "fmt"

func main() {
	var i interface{} = 10
	fmt.Println(i.(fmt.Formatter))
}
