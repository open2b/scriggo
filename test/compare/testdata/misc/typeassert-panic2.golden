panic: interface conversion: int is not fmt.Formatter